src/hooks/lastBlock/lib/getValidationData.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { ApiPromise } from '@polkadot/api';
+import { SignedBlock } from '@polkadot/types/interfaces';
+import { getValidationData, validationDataDataType } from './getValidationData';
+
+const rawValidationData = 'mockedRawValidationData';
+const mockJsonValidationData = 'result';
+const parachainBlock = {
+  block: { header: { hash: 'hash' } },
+} as unknown as SignedBlock;
+
+export const getMockApiPromise = (): ApiPromise =>
+  ({
+    query: {
+      parachainSystem: {
+        validationData: jest.fn(() => rawValidationData),
+      },
+    },
+    derive: {
+      chain: {
+        subscribeNewBlocks: jest.fn(),
+      },
+    },
+    createType: jest.fn(() => {
+      return {
+        isSome: true,
+        unwrap: () => ({ mockJsonValidationData }),
+      };
+    }),
+    at: jest.fn(() => getMockApiPromise()),
+  } as unknown as ApiPromise);
+
+describe('getValidationData', () => {
+  let mockApiInstance: ApiPromise;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    mockApiInstance = getMockApiPromise();
+  });
+
+  it('gets validation data from parachain block head', async () => {
+    const jsonValidationData = await getValidationData(
+      mockApiInstance,
+      parachainBlock
+    );
+
+    expect(jsonValidationData).toEqual({ mockJsonValidationData });
+    expect(mockApiInstance.at).toHaveBeenCalled();
+    expect(mockApiInstance.createType).toHaveBeenCalledWith(
+      validationDataDataType,
+      rawValidationData
+    );
+  });
+
+  it(`doesn't get validation data from parachain block head`, async () => {
+    (mockApiInstance.createType as jest.Mock).mockReturnValueOnce({
```

Comment:
Alternative to this would be using `as unknown as jest.Mocked<ApiPromise>` at the time of the mock setup. Then you can mock the return value as follows:

```
mockApiInstance.createType.mockReturnValueOnce({
      isSome: false,
    } as any as Codec);
```